src/factory-projects-manager.ts

```diff
@@ -39,24 +39,33 @@ export class FactoryTheiaClient {
         try {
             factory = await che.factory.getById(factoryId);
         } catch (e) {
-            theia.window.showErrorMessage('Unable to get factory. ' + e.message);
+            theia.window.showErrorMessage(`Unable to get factory. ${e.message}`);
             return;
         }
 
-        const cloneCommandList = await this.extractCloneCommands(factory);
-        const onProjectsImportedCommandList = this.extractOnProjectsImportedCommands(factory);
+        // Clone Factory projects
+        const cloneCommands = await this.getCloneCommands(factory);
+        if (cloneCommands.length > 0) {

```

Comment:
I would rather do this check inside the method execute

Reply:
agree